src/category_theory/abelian/additive_functor.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2021 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import category_theory.preadditive
+import category_theory.abelian.basic
+
+/-!
+# Additive Functors
+
+A functor between two preadditive categories is called *additive*
+provided that the induced map on hom types is a morphism of abelian
+groups.
+
+# Implementation details
+
+`functor.additive` is a `Prop`-valued class, defined by saying that
+for every two objects `X` and `Y`, there exists a morphism of additive
+groups `f : (X ⟶ Y) → (F.obj X ⟶ F.obj Y)` whose underlying function
+agrees with `F.map`.
+
+To construct an instance of `functor.additive G` from proofs that
+`G.map` sends `0` to `0` and is compatible with addition of morphisms,
+use `functor.additive.of_is_hom`.
+
+# Projects (in the case of abelian categories):
+
+- Prove that an additive functor preserves finite biproducts
+- Prove that a functor is additive it it preserves finite biproducts
+-/
+
+namespace category_theory
+
+/-- A functor `F` is additive provided `F.map` is an additive homomorphism. -/
+class functor.additive {C D : Type*} [category C] [category D]
+  [preadditive C] [preadditive D] (F : C ⥤ D) : Prop :=
+(exists_hom' : ∀ (X Y : C), ∃ f : (X ⟶ Y) →+ (F.obj X ⟶ F.obj Y),
+  ∀ g : X ⟶ Y, F.map g = f g)
```

Comment:
Hmm, isn't this making the same mistake as `class is_add_monoid_hom (f : ...)`, which associates a class with a value, meaning that simp lemmas never fire for it?

Reply:
I think it's okay (see sanity check example in latest commit).